src/Airship/Internal/Decision.hs

```diff
@@ -131,6 +131,32 @@ writeCacheTags Resource{..} = lift $ do
        Nothing -> return ()
        Just d  -> addResponseHeader ("Last-Modified", utcTimeToRfc1123 d)
 
+-- | Immediately halts processing with the provided 'Status' code. The response body is
+-- | optionally set based on the error responses provided by the application and resource.
+-- | The contents of the 'Webmachine''s response body will be streamed back to the client.
+-- | This is a shortcut for constructing a 'Response' with 'getResponseHeaders' and 'getResponseBody'
+-- | and passing that response to 'finishWith'.
+haltWithErrorResponse :: Monad m => HTTP.Status -> Resource m -> Webmachine m a
+haltWithErrorResponse status Resource{..} = do
```

Comment:
This suggestion might be a pain.

One small problem with having something different to `halt` is that consumers will not get this functionally for free (if they're naughty and use `halt` that is), as well as you having to change all of the `Decision.hs` functions manually.

One suggestion which you should feel _completely_ free to ignore would be to do (something like) inspect the `Response` in `resourceToWaiT` and if no body is set call `errorResponse`. The danger I see is if you really really want to halt with a status but no body and also have an error handler for the same status. Just an idea, I'm curious if anyone has an opinion on that one.